packages/hardhat-core/src/internal/vendor/await-semaphore/index.ts

```diff
@@ -64,3 +67,154 @@ export class Mutex extends Semaphore {
     super(1);
   }
 }
+
+export class MultiProcessMutex {
+  // Logic explanation: the fs.writeFile function, when used with the wx+ flag, performs an atomic operation to create a file.
```

Comment:
Why `wx+` and not just `wx`? Also, can we include a reference here? I don't see this mentioned in node docs.

Reply:
It's quite easy to find ion the fs nodeJS docs: https://nodejs.org/api/fs.html

<img width="403" alt="image" src="https://github.com/NomicFoundation/hardhat/assets/18092467/b5a31a79-1528-4d3f-bc54-fa3e5cebbfbc">